app/src/main/java/org/simple/clinic/widgets/qrcodescanner/QrCodeAnalyzer.kt

```diff
@@ -0,0 +1,74 @@
+package org.simple.clinic.widgets.qrcodescanner
+
+import android.graphics.ImageFormat.YUV_420_888
+import android.graphics.ImageFormat.YUV_422_888
+import android.graphics.ImageFormat.YUV_444_888
+import android.os.Build
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageProxy
+import com.google.zxing.BarcodeFormat
+import com.google.zxing.BinaryBitmap
+import com.google.zxing.DecodeHintType
+import com.google.zxing.MultiFormatReader
+import com.google.zxing.NotFoundException
+import com.google.zxing.PlanarYUVLuminanceSource
+import com.google.zxing.common.HybridBinarizer
+import java.nio.ByteBuffer
+
+private typealias OnQrCodeDetected = (String) -> Unit
+
+class QrCodeAnalyzer(
```

Comment:
Do you think we should rename this based on the method used to analyze the image?

Ex: `ZxingQrCodeAnalyzer`.